backend/cmm_helpers.mli

```diff
@@ -1096,17 +1110,33 @@ val unboxed_mutable_float32_unboxed_product_array_set :
 val unboxed_int32_array_ref :
   expression -> expression -> Debuginfo.t -> expression
 
+(* XXX mshinwell: are the sign extension semantics for the following two
+   functions reasonable? *)
```

Comment:
I think your current specification is reasonable, but I'd argue that it would be nice to have the property that the int32s are stored sign-extended. This links to my comment above about marshalling of arrays of unboxed float32s: I think it's better to have and enforce a canonical representation in those cases (as long as it's not too costly), to avoid surprising behaviours.

Reply:
I agree given that the store functions always write a sign-extended 64 bit word. But having the same semantics as float32 (undefined high bits) might make it easier to transition to tightly packing 32-bit fields in the future.